### PR TITLE
The Filter Chain should be `iterable`

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes - for other components, kind of.
| BC Break      | maybe
| New Feature   | yes

### Description

Numerous components iterate over the filter chain even though it is not iterable - `laminas-form`, `laminas-inputfilter` to name 2

The class is marked soft final, so I'm unsure if this is a BC break or not.

Also bumps dev dependencies and minimum version of `StdLib` to take advantage of better types

### CODE REVIEW

File: src/FilterChain.php [hunk 1]
```diff
@@ -225,13 +227,19 @@ public function getFilters()
      *
      * @param  mixed $value
      * @return mixed
+     * @psalm-suppress MixedAssignment values are always mixed
      */
     public function filter($value)
     {
-        $chain = clone $this->filters;
-
+        $chain         = clone $this->filters;
```
Comment (Ocramius) [inline src/FilterChain.php]: Couldn't this become `foreach ($this ...`? 🤔

**Review by Ocramius — CHANGES_REQUESTED** [review]
I like the idea here - unsure if valid architecturally, since the internal details shouldn't be exposed 🤔

**Comment by froschdesign** [comment]
@gsteel 
Is the same needed for the validator chain in laminas-validator?

https://github.com/laminas/laminas-validator/blob/7f15f8136f913d17ba39ee595637e1e8348b1aac/src/ValidatorChain.php#L20-L22

**Comment by gsteel** [comment]
> Is the same needed for the validator chain in laminas-validator?

@froschdesign 

Yes!

**Review by Ocramius — APPROVED** [review]
🚢  thanks @gsteel!

**Comment by gsteel** [comment]
> I like the idea here - unsure if valid architecturally, since the internal details shouldn't be exposed

Well the filter chain (and validator chain) are well known as a list of filters that are executed in order. Would it be better for `getIterator()` to return a new `ArrayIterator(iterator_to_array($this->filters))` rather than returning a clone of the `PriorityQueue`?

**Comment by Ocramius** [comment]
> rather than returning a clone of the `PriorityQueue`?

That's an implementation detail: the consumer just knows it's a `Traversable`

**Pull request merged**